core/modules/replica/Application.cc

```diff
@@ -48,17 +48,21 @@ Application::Application(int argc,
                          string const& description,
                          bool const injectDatabaseOptions,
                          bool const boostProtobufVersionCheck,
-                         bool const enableServiceProvider)
+                         bool const enableServiceProvider,
+                         bool const injectXrootdOptions)
     :   _injectDatabaseOptions    (injectDatabaseOptions),
         _boostProtobufVersionCheck(boostProtobufVersionCheck),
         _enableServiceProvider    (enableServiceProvider),
+        _injectXrootdOptions      (injectXrootdOptions),
         _parser   (argc,argv, description),
         _debugFlag(false),
-        _config   ("file:replication.cfg"),
+        _config   ("mysql://qsreplica@localhost:3306/qservReplica"),
```

Comment:
Actually, it's implemented exactly like you say. The class `Application` adds the optional command line parameter:
```
--config=<config-url>
```
What I'm doing in the code above is just setting some reasonable default. In the previous version of the code (before it got refactored) the default was set to some file in the user's PWD. After eliminating the file-based configuration support, I had to set it to something else. In the new version of the code the default was made to simplify things for the mono-node integration tests, and it's based on the following assumptions:
* the MySQL service is run locally
* the service is shared by Qserv (`czar`, workers) and the Replication/Ingest system
* using the default MySQL port
* using the default account for the Replication system

Reply:
I think the default should only be defined in one location (say Application.h) so it's something like ```_config(Application::getDefaultMysql())``` or ```_config(Application::DefaultMysql)```.

Reply:
I have a better idea. Right now the very same default is specified in two classes:
```
% fgrep 'mysql://qsreplica@localhost:3306/qservReplica' core/modules/replica/*
core/modules/replica/Application.cc:        _config   ("mysql://qsreplica@localhost:3306/qservReplica"),
core/modules/replica/ConfigApp.cc:        _configUrl("mysql://qsreplica@localhost:3306/qservReplica")
```
While the second is a sub-class of the first one. The derived class ignores a fact that the base one already defined option:
```
--config=<url>
```
And defines the same option. This is clearly a mistake. I'm going to fix that by extending the base class with method:
```
std::string Application::configUrl() const;
```
This will eliminate the duplicate.

Thank you very much for bringing this up!